### PR TITLE
fix: Fix listener stopping on dispose

### PR DESCRIPTION
**Pull request by mark-kascheev**
## Related tasks

Closes https://github.com/surfstudio/flutter-otp-autofill/issues/38

## Changes
Missing call of ```unRegisterBroadcastReceivers()```

## Checklist for self-check
- [x] Commits and PRs have been filed according to [the rules on the project](https://github.com/surfstudio/surf-flutter-app-template#workflow-in-a-repository).
- [x] The author is marked as an assigne and assigned mandatory reviewers.
- [x] Required labels marked
- [x] Specified related tasks and/or related PRs.
- [x] Specified Changes.
- [ ] Attached videos/screenshots demonstrating the fix/feature.
- [x] All unspecified fields in the PR description deleted.
- [ ] New code covered by tests.

## Checklist for reviewers
- [x] CI passed successfully _(with a green check mark)_.
- [x] PR is atomic, by volume no more than 400 (+-) corrected lines (not including codogen).

Design:
- [x] System design corresponds to the agreements on structure and architecture on the project.
- [x] The code is decomposed into necessary and sufficient components.

Functionality:
- [x] The code solves the problem.
- [x] Any changes to the user interface are reasonable and look good.

Complexity:
- [x] The code is clear, easy to read, functions are small, no more than 50 lines.
- [x] The logic is not overcomplicated, there is no overengineering (no code sections that may be needed in the future, but no one knows about it).

Tests:
- [ ] Updated or added tests for mandatory components.
- [ ] The tests are correct, helpful, and well designed/developed.

Naming:
- [x] The naming of variables, methods, classes and other components is understandable.

Comments:
- [ ] The comments are understandable and helpful.

Documentation:
- [x] All labels are correct
- [ ] Technical documentation updated (after approval, updates last reviewer).

### CODE REVIEW

File: .fvm/flutter_sdk [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/parshonok/fvm/versions/3.7.6
```
Comment (dshevchenkoo) [inline .fvm/flutter_sdk:1]: add to gitignore

File: android/src/main/kotlin/ru/surfstudio/otp_autofill/OTPPlugin.kt [hunk 2]
```diff
@@ -113,6 +113,7 @@ public class OTPPlugin : FlutterPlugin, MethodCallHandler, PluginRegistry.Activi
 
     override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
         channel.setMethodCallHandler(null)
+        unRegisterBroadcastReceivers()
```
Comment (dshevchenkoo) [inline android/src/main/kotlin/ru/surfstudio/otp_autofill/OTPPlugin.kt:116]: add comment

File: example/.fvm/flutter_sdk [hunk 3]
```diff
@@ -0,0 +1 @@
+/Users/parshonok/fvm/versions/3.7.6
```
Comment (mpkander) [inline example/.fvm/flutter_sdk:1]: You also need to delete the file

File: pubspec.yaml [hunk 4]
```diff
@@ -17,7 +17,7 @@ dev_dependencies:
   surf_lint_rules: ^2.0.0
 
 environment:
-  sdk: ">=2.12.0 <3.0.0"
+  sdk: ">=2.18.0 <3.0.0"
```
Comment (mpkander) [inline pubspec.yaml:20]: Is it necessary to upgrade SDK?

**Review by mpkander — CHANGES_REQUESTED** [review]

File: pubspec.yaml (hunk 4 above)
Comment (plasticfiresam) [reply pubspec.yaml]: Nope, fixed

File: example/.fvm/flutter_sdk (hunk 3 above)
Comment (plasticfiresam) [reply example/.fvm/flutter_sdk]: Deleted

**Review by maxim-sysoev — APPROVED** [review]

**Review by mpkander — APPROVED** [review]
LGTM

**Pull request merged**